mrmustard/lab_dev/circuit_components.py

```diff
@@ -151,11 +150,32 @@ def bargmann(self) -> tuple:
         r"""
         The Bargmann parametrization of this circuit component, if available.
         """
-        if not isinstance(self.representation, Bargmann):
-            raise ValueError(
+        try:
+            return self.representation.triple
+        except AttributeError as e:
+            raise AttributeError(
                 f"Cannot compute triple from representation of type ``{self.representation.__class__.__qualname__}``."
-            )
-        return self.representation.triple
+            ) from e
+
+    def quadrature(self) -> tuple | ComplexTensor:
+        r"""
+        The quadrature representation of this circuit component.
+        """
+        from mrmustard.lab_dev.circuit_components_utils import (  # pylint: disable=import-outside-toplevel
+            BtoQMap,
+        )
```

Comment:
Why do we have to do the import here?

Reply:
Should we be following the same structure as to_fock, and put a converter in the physics module?

Reply:
It's there to make it local to the scope, to avoid a circular import (BtoQMap is a CircuitComponent and needs to import CircuitComponent to inherit from it).

Reply:
And no, if anything I want to move away to having converters separate from the representations.

Reply:
yes, for two reasons: 1) we bring functionality together: if I'm in a representation I don't need to jump somewhere else to see how the conversion is done and 2) we avoid a few if statements each time we convert because only the appropriate conversion procedure is followed.